packages/payment-detection/src/eth/input-data.ts

```diff
@@ -8,10 +8,37 @@ import Utils from '@requestnetwork/utils';
 import PaymentReferenceCalculator from '../payment-reference-calculator';
 
 import EthInputDataInfoRetriever from './info-retriever';
+import EthProxyInputDataInfoRetriever from './proxy-info-retriever';
 
 const bigNumber: any = require('bn.js');
 const supportedNetworks = ['mainnet', 'rinkeby', 'private'];
 
+interface IProxyContractByVersionByNetwork {
+  [version: string]: {
+    [network: string]: { address: string; creationBlockNumber: number };
+  };
+}
+
+const firstVersion = {
+  mainnet: {
+    address: '0x37a8f5f64f2a84f2377481537f04d2a59c9f59b6',
+    creationBlockNumber: 9466832,
+  },
+  private: {
+    address: '0xf204a4ef082f5c04bb89f7d5e6568b796096735a',
+    creationBlockNumber: 0,
+  },
+  rinkeby: {
+    address: '0x9c6c7817e3679c4b3f9ef9486001eae5aaed25ff',
+    creationBlockNumber: 5955681,
+  },
+};
+
+const PROXY_CONTRACT_ADDRESS_BY_VERSION_BY_NETWORK: IProxyContractByVersionByNetwork = {
```

Comment:
Would be nice to have a comment here too.

Maybe `PROXY_CONTRACT_ADDRESS_MAP` is clear enough? Feels a bit too verbose (nitpicking, you can leave like this if you want).